media_driver/linux/common/codec/ddi/media_ddi_decode_hevc.cpp

```diff
@@ -433,6 +433,14 @@ VAStatus DdiDecodeHEVC::RenderPicture(
         }
         case VASubsetsParameterBufferType:
         {
+
+            if (m_ddiDecodeCtx->DecodeParams.m_subsetParams == nullptr) {
+                m_ddiDecodeCtx->DecodeParams.m_subsetParams = MOS_AllocAndZeroMemory(sizeof(CODEC_HEVC_SUBSET_PARAMS));
```

Comment:
Better not move init here. it will cause some issue. In hal level,  umd will copy m_subsetParams again to another array and use the array for long format decode. If the array is nullptr, it will return error. 
check function 'InitSliceTileParams' in codechal_decode_hevc_long_g12.cpp

Reply:
`InitSliceTileParams` is called when `m_isSeparateTileDecoding || m_isRealTile` is true
1. for read tile decoding,  user has already provided a valid subset buffer by calling `vaRenderPicture`, so `InitSliceTileParams` should return MOS_STATUS_SUCCESS
2. If separate tile decoding,  may user not provide subset buffer?

Reply:
@Jexu Thanks for the comment, I added code to handle the case of m_subsetParams=NULL, could you help to review it again?

Reply:
done！it is fine.